xwiki-platform-core/xwiki-platform-web/src/main/webapp/resources/uicomponents/viewers/comments.js

```diff
@@ -63,21 +63,62 @@ viewers.Comments = Class.create({
   },
   /** Enhance the Comments UI with JS behaviors. */
   startup : function () {
-    if ($("commentform")) {
-      this.form = $("commentform").up("form");
-    } else {
-      this.form = undefined;
-    }
     this.loadIDs();
     this.addReplyListener();
-    this.addSubmitListener(this.form);
-    this.addCancelListener();
     this.addEditListener();
-    this.reloadEditor({
-      callback: function () {
-        this.addPreview(this.form);
-      }.bind(this)
-    });
+    // replaces the comment button with the form on the first click.
+    $("OpenCommentForm").observe('click', function (event) {
+      event.stop();
+      this.toggleFormHidden(function () {
+        this.reloadEditor({
+          callback: function () {
+            this.getForm();
+            this.addSubmitListener(this.form);
+            this.addCancelListener();
+            this.addPreview(this.form);
+          }.bind(this)
+        });
+      }.bind(this));
+    }.bind(this));
+  },
+  // lazily load the form element
+  getForm: function () {
+    this.form = this.form || undefined;
+    var commentform = $("commentform");
+    if (this.form === undefined && commentform) {
+      this.form = commentform.up("form");
+    }
+  },
+  // the form must be loaded only once.
+  formToggled: false,
+  // load the form only once, when it is first needed
+  // the next calls to this method does nothing but calling the callback
+  // function parameter.
+  toggleFormHidden: function (callback) {
+    if (!this.formToggled) {
+      require(['jquery'], function ($) {
+        var url = window.location.href;
+        $.ajax({
```

Comment:
I don't see the benefit of loading the comment form (without the WYSIWYG editor) asynchronously. The form is just a few lines of HTML and generating them takes very little. Why not simply have the comment form hidden inside the comments tab and make the comment button just toggle its visibility. It feels a bit too much to have a separate HTTP request to take the comments tab content without the comment form and then make another HTTP request to take the form HTML without the editor. Overall it takes more time.